cmd/dataprotection/main.go

```diff
@@ -0,0 +1,321 @@
+
+	// +kubebuilder:scaffold:imports
+
+	dpv1alpha1 "github.com/apecloud/kubeblocks/apis/dataprotection/v1alpha1"
+	storagev1alpha1 "github.com/apecloud/kubeblocks/apis/storage/v1alpha1"
+	dpcontrollers "github.com/apecloud/kubeblocks/controllers/dataprotection"
+	"github.com/apecloud/kubeblocks/internal/constant"
+	intctrlutil "github.com/apecloud/kubeblocks/internal/controllerutil"
+	viper "github.com/apecloud/kubeblocks/internal/viperx"
+)
+
+// added lease.coordination.k8s.io for leader election
+// +kubebuilder:rbac:groups=coordination.k8s.io,resources=leases,verbs=get;list;watch;create;update;patch
+
+const (
+	appName = "kubeblocks"
+)
+
+type flagName string
+
+const (
+	probeAddrFlagKey     flagName = "health-probe-bind-address"
+	metricsAddrFlagKey   flagName = "metrics-bind-address"
+	leaderElectFlagKey   flagName = "leader-elect"
+	leaderElectIDFlagKey flagName = "leader-elect-id"
+)
+
+var (
+	scheme   = runtime.NewScheme()
+	setupLog = ctrl.Log.WithName("setup")
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(dpv1alpha1.AddToScheme(scheme))
+	utilruntime.Must(snapshotv1.AddToScheme(scheme))
+	utilruntime.Must(snapshotv1beta1.AddToScheme(scheme))
+	utilruntime.Must(storagev1alpha1.AddToScheme(scheme))
+	// +kubebuilder:scaffold:scheme
+
+	viper.SetConfigName("config")                          // name of config file (without extension)
+	viper.SetConfigType("yaml")                            // REQUIRED if the config file does not have the extension in the name
+	viper.AddConfigPath(fmt.Sprintf("/etc/%s/", appName))  // path to look for the config file in
+	viper.AddConfigPath(fmt.Sprintf("$HOME/.%s", appName)) // call multiple times to append search path
+	viper.AddConfigPath(".")                               // optionally look for config in the working directory
+	viper.AutomaticEnv()
+
+	viper.SetDefault(constant.CfgKeyCtrlrReconcileRetryDurationMS, 1000)
+	viper.SetDefault("CERT_DIR", "/tmp/k8s-webhook-server/serving-certs")
+	viper.SetDefault("VOLUMESNAPSHOT", false)
+	viper.SetDefault("VOLUMESNAPSHOT_API_BETA", false)
+	viper.SetDefault(constant.KBToolsImage, "apecloud/kubeblocks-tools:latest")
+	viper.SetDefault("KUBEBLOCKS_SERVICEACCOUNT_NAME", "kubeblocks")
+	viper.SetDefault(constant.CfgKeyCtrlrMgrNS, "default")
+	viper.SetDefault(constant.KubernetesClusterDomainEnv, constant.DefaultDNSDomain)
+}
+
+func main() {
+	var (
+		metricsAddr            string
+		enableLeaderElection   bool
+		enableLeaderElectionID string
+		probeAddr              string
+	)
+
+	flag.String(metricsAddrFlagKey.String(), ":8080", "The address the metric endpoint binds to.")
+	flag.String(probeAddrFlagKey.String(), ":8081", "The address the probe endpoint binds to.")
+	flag.Bool(leaderElectFlagKey.String(), false,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.")
+
+	flag.String(leaderElectIDFlagKey.String(), "abd03fda",
```

Comment:
It should be inconsistent with the electId of kb

Reply:
The electId of manager is `001c317f`, it does not consistent.